### DISCUSSION TITLE
Linking from one .md file to another

### DISCUSSION THREAD

**Discussion by ba58smith** [community/community#22595] [New to GitHub]
<p>I’m working on two different documents for the same project: Overview.md and Getting Started.md. I’m working on them in my local repository, in a new branch just for adding documentation. I want to link from each of them to the other one, but nothing I put into the url part of the link (the second part, in parentheses) seems to work. I’ve tried:</p>
<p>[click here for the Getting Started documentation](Getting Started.md)</p>
<p>and</p>
<p>[click here for the Getting Started documentation](myProjectName/Getting Started.md)</p>
<p>What I really want is the URL that will be correct once I finally commit the document to the Master branch of the repo.</p>
<p>(BTW, does it matter that I have a space in my document name? It didn’t work when I tried to link to the other document, either, and that document has no space in the name.)</p>
<p>Thanks!</p>

**Comment by mpboom** [comment, marked as answer]
<p>Well, if your goal is that it will work when you finally commit it to the master branch, just put this URL in:</p>
<pre><code>https://github.com/&lt;username&gt;/&lt;repo name&gt;/blob/master/&lt;file path to file&gt;
</code></pre>

**Comment by ba58smith** [comment]
<p>Yeah, I guess that’s really all I care about. I was hoping to be able to see it work in the working branch before the commit to Master, but I guess that’s not really all that important. Thanks, <a class="mention" href="/mpboom">@mpboom</a>.</p>

**Comment by lee-dohm** [comment]
<p>If they’re in the root of your repo, then you should only need to use the name of the file with nothing else. GitHub rewrites the URLs so that it should work no matter what branch you’re on. See the <a href="https://help.github.com/articles/basic-writing-and-formatting-syntax/#relative-links" rel="nofollow noopener">relative links help documentation</a> for a full explanation.</p>
<p>But yes, having spaces in the file name does cause a problem because, <a href="https://spec.commonmark.org/0.28/#link-destination" rel="nofollow noopener">according to the Markdown specification</a>, link destinations are not allowed to have spaces in them. You’ll notice that I was able to <a href="https://github.com/lee-dohm/test-repo/blob/spacey-file-name/file%20with%20spaces%20in%20the%20name.md" rel="nofollow noopener">easily link to the file with no spaces in the name here</a>. But when trying to <a href="https://github.com/lee-dohm/test-repo/blob/spacey-file-name/non-spacey-file-name.md" rel="nofollow noopener">link back to the file with spaces</a>, it doesn’t work.</p>
<p>I hope that helps!</p>

**Comment by ba58smith** [comment]
<p>Got it! Thanks for the clarification!</p>

**Comment by moopsiegoo** [comment]
<p>I found a solution!!</p>
<p>You have to replace the spaces with %20 and it worked for me! Here both .md files were in the same folder.</p>
<p>Just do something along the lines </p>
<pre><code>Continue on to [Chapter 2: Hello World app](Chapter%202:%20Hello%20World%20app.md).
</code></pre>
<p>Hope this helps :slight_smile:</p>

**Comment by ba58smith** [comment]
<p>Thanks for that! I have since moved all of my docs to the wiki, and it allows you to connect pages to each other (all within the wiki) like this. My other document is called “Getting Started”, and although I feel like it’s probably stored as a separate file, in the wiki system, it’s referred to as a “page”. At any rate… pages / documents within the same wiki can be linked like this:</p>
<p><a>click here for the Getting Started documentation</a>.</p>
